pkg/newlog/cmd/newlogd.go

```diff
@@ -0,0 +1,1281 @@
+}
+
+// Handles and combine loguploader side of Newlog metrics
+func handleUploadMetricsImp(ctxArg interface{}, key string, statusArg interface{}) {
+	status := statusArg.(types.NewlogMetrics)
+	logmetrics.TotalBytesUpload = status.TotalBytesUpload
+	logmetrics.Num4xxResponses = status.Num4xxResponses
+	logmetrics.Latency.MinUploadMsec = status.Latency.MinUploadMsec
+	logmetrics.Latency.MaxUploadMsec = status.Latency.MaxUploadMsec
+	logmetrics.Latency.AvgUploadMsec = status.Latency.AvgUploadMsec
+	logmetrics.Latency.CurrUploadMsec = status.Latency.CurrUploadMsec
+
+	logmetrics.CurrUploadIntvSec = status.CurrUploadIntvSec
+
+	logmetrics.ServerStats.CurrCPULoadPCT = status.ServerStats.CurrCPULoadPCT
+	logmetrics.ServerStats.AvgCPULoadPCT = status.ServerStats.AvgCPULoadPCT
+	logmetrics.ServerStats.CurrProcessMsec = status.ServerStats.CurrProcessMsec
+	logmetrics.ServerStats.AvgProcessMsec = status.ServerStats.AvgProcessMsec
+
+	// loguplader signal to newlogd on upload fail status
+	logmetrics.FailedToSend = status.FailedToSend
+	logmetrics.FailSentStartTime = status.FailSentStartTime
+
+	logmetrics.DevMetrics.NumGZipFilesSent = status.DevMetrics.NumGZipFilesSent
+	logmetrics.DevMetrics.NumGzipFileInDir = status.DevMetrics.NumGzipFileInDir
+	logmetrics.DevMetrics.NumGZipFileRetry = status.DevMetrics.NumGZipFileRetry
+	logmetrics.DevMetrics.RecentUploadTimestamp = status.DevMetrics.RecentUploadTimestamp
+	logmetrics.DevMetrics.LastGZipFileSendTime = status.DevMetrics.LastGZipFileSendTime
+	logmetrics.DevMetrics.NumGZipFileKeptLocal = status.DevMetrics.NumGZipFileKeptLocal
+
+	logmetrics.AppMetrics.NumGZipFilesSent = status.AppMetrics.NumGZipFilesSent
+	logmetrics.AppMetrics.NumGzipFileInDir = status.AppMetrics.NumGzipFileInDir
+	logmetrics.AppMetrics.NumGZipFileRetry = status.AppMetrics.NumGZipFileRetry
+	logmetrics.AppMetrics.RecentUploadTimestamp = status.AppMetrics.RecentUploadTimestamp
+	logmetrics.AppMetrics.LastGZipFileSendTime = status.AppMetrics.LastGZipFileSendTime
+	logmetrics.AppMetrics.NumGZipFileKeptLocal = status.AppMetrics.NumGZipFileKeptLocal
+
+	log.Tracef("newlogd handleUploadMetricsModify changed to %+v", status)
+}
+
+// Handles UUID change from process client
+func handleOnboardStatusCreate(ctxArg interface{}, key string, statusArg interface{}) {
+	handleOnboardStatusImp(ctxArg, key, statusArg)
+}
+
+// Handles UUID change from process client
+func handleOnboardStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handleOnboardStatusImp(ctxArg, key, statusArg)
+}
+
+// Handles UUID change from process client
+func handleOnboardStatusImp(ctxArg interface{}, key string, statusArg interface{}) {
+	status := statusArg.(types.OnboardingStatus)
+	if cmp.Equal(devMetaData.uuid, status.DeviceUUID.String()) {
+		log.Tracef("newlogd handleOnboardStatusModify no change to %s", devMetaData.uuid)
+		return
+	}
+	devMetaData.uuid = status.DeviceUUID.String()
+	logmetaData = formatAndGetMeta("")
```

Comment:
What happens to logmetaData? Does it have to be a global variable?

Reply:
here we just format the device log related meta-data (non-app, so no app-uuid string). i don't have context structure in newlogd since it's a simple one file program.

Reply:
I don't understand from the local context what logmetaData is for. Makes it harder to understand because it is a global variable. I guess I have to grep for it.

Reply:
ok. i'll put some comments there. it's a string composed of device-uuid, eve-release and partition number. so the device subscription calls can update the fields, it's being used for later every new logfile to include this metadata.